optuna/visualization/_utils.py

```diff
@@ -99,3 +103,19 @@ def _get_param_values(trials: List[FrozenTrial], p_name: str) -> List[Any]:
     if _is_numerical(trials, p_name):
         return values
     return list(map(str, values))
+
+
+def _filter_nonfinite(trials: List[FrozenTrial]) -> List[FrozenTrial]:
```

Comment:
I realised that we need to care about transformed value by `target` that is used in visualization methods. I think we can extend this function by adding `target` argument.


```suggestion
def _filter_nonfinite(trials: List[FrozenTrial], target: Optional[Callable[[FrozenTrial], float]] = None) -> List[FrozenTrial]:
```

Reply:
Good catch! So i guess we should be doing transformation like
```python
if target is None:
  value = trial.value
else:
  value = target(trial)
```
before doing checks with `isfinite`.

Reply:
I think so too!

Reply:
Done in f0a8618d6137ea870f1477880ad7ae847651db38. CC @HideakiImamura, this patch removes explicit handling of multi-objective studies, since those are [required to specify target](https://github.com/optuna/optuna/blob/f072d2cd8d4e8cd35244cf65ab720fd7366aad7e/optuna/visualization/_utils.py#L57-L61) which selects one of objectives to plot. From our point of view, we are always dealing with single objective.

Reply:
Sounds reasonable. Thanks for the update.

Reply:
How about leaving a comment that states this utility only supports the single-objective optimization?

Reply:
Comment added as part of 817cef4700471fc41fbd451075e91ed81989cede :^)